package.json

```diff
@@ -20,7 +20,7 @@
     "> 1%"
   ],
   "dependencies": {
-    "@automattic/muriel-base-components": "file:packages/muriel-base-components",
```

Comment:
With the removal of internal `file:` dependencies, we may be able to recover our `npm install` cached Docker layer that we lost recently:

https://github.com/Automattic/wp-calypso/pull/27356#discussion_r220987034

🤔 That said, I'm not sure how that will play with `lerna`. Packages are attemted to be resolved locally first from the monorepo ad `packages/*`? It seems that we would have a the same problem and no local packages would be found so the published versions would be pulled 😞 

Worth flagging here, we can play with this in a follow up.

Reply:
With lerna, it first pulls the published versions, then overwrites them with the symlink to the local version.

Reply:
This change doesn't work as expected for me. `wp-calypso/node_modules/@automattic/tree-select` is the published version of the package. `lerna bootstrap` doesn't overwrite it with a symlink to local directory.

Lerna seems to do the symlinking only for packages managed by Lerna. And the root app is not one of them. Only when the packages in `wp-calypso/packages` depend on each other, the cross-dependencies in `/packages/**/node_modules` are cross-symlinked if the local package satisifies the semvers.

If the root `package.json` contains `file:*` links, running `lerna bootstrap` is [identical](https://github.com/lerna/lerna/blob/eae5619840c385fdebceadfca8f280a515ddd445/commands/bootstrap/index.js#L109-L111) to `npm install` and therefore redundant. All dependencies in the subpackages, external and internal, will be hoisted in the root `wp-calypso/node_modules` and correctly satisfied from there.

Maybe even the Docker layer should survive? In addition to the root `package.json`, it now depends on all the `packages/**/package.json` files, too. And some directories in the root `node_modules` folder are now symlinks. Is any of that an obstacle to Docker caching?